web/layouts/404.html

```diff
@@ -4,10 +4,17 @@ <h2>Page not found!</h2>
 
 <p>We are sorry, but this page does not exist.
 <script>
-    if (document.location.toString().includes("/master/")) {
-        document.write("This page has been <a href=\"")
-        document.write(document.location.toString().replace("/master/", "/nightly/"))
-        document.write("\">moved</a>.")
+    var loc = document.location.toString();
+    if (loc.includes("/master/") || loc.includes("/index.html")) {
+        newLocation = document.location.toString()
```

Comment:
I think  instead of `document.location.toString()` you can use `window.location.href` at all places.